addons/mail/models/mail_activity.py

```diff
@@ -374,6 +374,7 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
             DocumentModel = self.env[doc_model].with_user(access_rights_uid or self._uid)
             right = DocumentModel.check_access_rights(doc_operation, raise_exception=False)
             if right:
+                doc_ids.discard(0)  # in case it was used as a hack to signal no linked record
```

Comment:
Avoid processing an empty document list
```suggestion
            doc_ids.discard(0)  # in case it was used as a hack to signal no linked record
            if right and doc_ids:
```

Reply:
done, thanks for the review!